### PR TITLE
Add support for int16 dtype regardless of 16x8 quantization usage

### PR DESCRIPTION
**Pull request by pjpratik**
As a feature request #56615, the _dtypes.int16 to be allowed even when 16x8 quantization is not used so that the custom ops returning int16 outputs can be benefitted.

### CODE REVIEW

**Review by sirakiin — APPROVED** [review]

**Comment by mihaimaruseac** [comment]
https://cbea.ms/git-commit/

**Comment by gbaned** [comment]
Hi @pjpratik Can you please resolve conflicts? Thank you!

**Comment by gbaned** [comment]
Hi @sirakiin Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @vamsimanchala Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @vamsimanchala Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @vamsimanchala Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @vamsimanchala Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @vamsimanchala Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @vamsimanchala Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @vamsimanchala Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @vamsimanchala Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @vamsimanchala Can you please review this PR ? Thank you!

File: tensorflow/lite/python/lite.py [hunk 1]
```diff
@@ -1086,6 +1086,9 @@ def _validate_inference_input_output_types(self, quant_mode):
       if quant_mode.is_post_training_int16x8_quantization():
         all_types = default_types + [_dtypes.int16]
       else:
+        all_types = default_types + [_dtypes.int8, _dtypes.uint8, _dtypes.int16]
+      if (self.inference_input_type not in all_types or
```
Comment (rascani) [inline tensorflow/lite/python/lite.py:1090]: This seems like a bad merge. I think you want to delete lines 1090-1092.

**Review by rascani — CHANGES_REQUESTED** [review]

File: tensorflow/lite/python/lite.py [hunk 2]
```diff
@@ -1086,6 +1086,9 @@ def _validate_inference_input_output_types(self, quant_mode):
...
+          self.inference_output_type not in all_types):
```
Comment (mihaimaruseac) [inline tensorflow/lite/python/lite.py:1091]:
```suggestion
```

**Review by mihaimaruseac — APPROVED** [review]

File: tensorflow/lite/python/lite.py [hunk 3]
```diff
@@ -1086,6 +1086,7 @@ def _validate_inference_input_output_types(self, quant_mode):
       if quant_mode.is_post_training_int16x8_quantization():
         all_types = default_types + [_dtypes.int16]
       else:
+        all_types = default_types + [_dtypes.int8, _dtypes.uint8, _dtypes.int16]
         all_types = default_types + [_dtypes.int8, _dtypes.uint8]
```
Comment (mihaimaruseac) [inline tensorflow/lite/python/lite.py:1090]:
```suggestion
        all_types = default_types + [_dtypes.int8, _dtypes.uint8, _dtypes.int16]
```

**Review by mihaimaruseac — APPROVED** [review]

File: tensorflow/lite/python/lite.py (hunk 1 above)
Comment (mihaimaruseac) [reply tensorflow/lite/python/lite.py]: Fixed via suggested edits

**Review by rascani — APPROVED** [review]

**Pull request merged**